File: algorithms/1_easy/M/03364_minimum_positive_sum_subarray/sliding_window.cpp
class Solution {
public:
    int minimumSumSubarray(vector<int>& nums, int l, int r) {
        const size_t len{nums.size()};
        int min_positive_sum{-1};
        const size_t min_width{static_cast<size_t>(l)};
        const size_t max_width{static_cast<size_t>(r)};

        for (size_t width{min_width}; width <= max_width; ++width) {
            int sum{0};
            size_t left{0};
            size_t right{width - 1};

            for (size_t i{left}; i <= right; ++i) {
                sum += nums[i];
            }

            while (right < len) {
                if (sum > 0 && (sum < min_positive_sum || min_positive_sum == -1)) {
                    min_positive_sum = sum;
                }

                right += 1;
                if (right < len) {
                    sum += nums[right];
                    sum -= nums[left];
                    left += 1;
                }
            }
        }

        return min_positive_sum;
    }
};

File: algorithms/1_easy/M/03364_minimum_positive_sum_subarray/sliding_window.py
class Solution:
    def minimumSumSubarray(self, nums: List[int], min_width: int, max_width: int) -> int:
        min_positive_sum = -1
        n = len(nums)

        for width in range(min_width, max_width + 1):
            left = 0
            right = width - 1
            curr_sum = sum(nums[left:right + 1])
    
            while right < n:
                if curr_sum > 0 and (curr_sum < min_positive_sum or min_positive_sum == -1):
                    min_positive_sum = curr_sum

                right += 1
                if right < n:
                    curr_sum += nums[right]
                    curr_sum -= nums[left]
                    left += 1

        return min_positive_sum
